py_ecc/bls/api.py

```diff
@@ -109,3 +113,48 @@ def verify_multiple(pubkeys: Sequence[BLSPubkey],
         return final_exponentiation == FQ12.one()
     except (ValidationError, ValueError, AssertionError):
         return False
+
+
+def verify_multiple_multiple(
+        signatures: Sequence[BLSSignature],
+        pubkeys_and_messages: Tuple[Sequence[BLSPubkey], Sequence[Hash32]],
+        domain: int)-> bool:
+    if len(signatures) != len(pubkeys_and_messages):
+        raise ValidationError(
+            "len(signatures) (%s) should be equal to len(pubkeys_and_messages) (%s)" % (
+                len(signatures), len(pubkeys_and_messages)
+            )
+        )
+    try:
+        random_ints = (1,) + tuple(2**randbelow(64) for _ in range(len(signatures) - 1))
+        o = FQ12.one()
+        for index, pm in enumerate(pubkeys_and_messages):
+            pubkeys, message_hashes = pm  # type: ignore
```

Comment:
We've been ensuring we add a comment explaining why any `# type: ignore` statements are present so it's clear why they exist.